mod/search/search_hooks.php

```diff
@@ -17,15 +17,18 @@
  */
 function search_objects_hook($hook, $type, $value, $params) {
 
+	$params['joins'] = (array) elgg_extract('joins', $params, array());
+	$params['wheres'] = (array) elgg_extract('wheres', $params, array());
+	
 	$db_prefix = elgg_get_config('dbprefix');
 
 	$join = "JOIN {$db_prefix}objects_entity oe ON e.guid = oe.guid";
-	$params['joins'] = array($join);
-	$fields = array('title', 'description');
+	array_unshift($params['joins'], $join);
```

Comment:
See the issue with annotations. Order of joins forces you to rejoin in case
you need the same table alias.
On Dec 21, 2015 10:41 PM, "Steve Clay" <notifications@github.com> wrote:

> In mod/search/search_hooks.php
> <https://github.com/Elgg/Elgg/pull/9252#discussion_r48195553>:
>
> >  	$db_prefix = elgg_get_config('dbprefix');
> >
> >  	$join = "JOIN {$db_prefix}objects_entity oe ON e.guid = oe.guid";
> > -	$params['joins'] = array($join);
> > -	$fields = array('title', 'description');
> > +	array_unshift($params['joins'], $join);
>
> Why unshift instead of push?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Elgg/Elgg/pull/9252/files#r48195553>.
>